utilities/realsense-test/src/Module.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <fstream>
+#include <iomanip>
+#include <yarp/os/LogStream.h>
+#include <BipedalLocomotion/RealSenseTest/Module.h>
+#include <BipedalLocomotion/ParametersHandler/YarpImplementation.h>
+#include <boost/make_shared.hpp>
+
+using namespace BipedalLocomotion;
+using namespace BipedalLocomotion::RealSenseTest;
+
+double Module::getPeriod()
+{
+    return m_dT;
+}
+
+bool Module::configure(yarp::os::ResourceFinder& rf)
+{
+    auto parametersHandler = std::make_shared<ParametersHandler::YarpImplementation>(rf);
+
+    if(!parametersHandler->getParameter("sampling_time", m_dT))
+    {
+        return false;
+    }
+
+    auto rsptr = parametersHandler->getGroup("REALSENSE").lock();
+    if (rsptr == nullptr)
+    {
+        yError() << "[Module::configure] Robot interface options is empty.";
+        return false;
+    }
+
+    rsDev = std::make_unique<Perception::Capture::RealSense>();
+    if (!rsDev->initialize(rsptr))
+    {
+        yError() << "[Module::configure] could not initialize realsense camera.";
+        return false;
+    }
+
+    std::cout << "[Module::configure] Starting the experiment." << std::endl;
+
+
+    pc = boost::make_shared<pcl::PointCloud<pcl::PointXYZRGB>>();    
```

Comment:
I think the macOS failure is related to the fact that in recent PCL versions `pcl::PointCloud<pcl::PointXYZRGB>::Ptr` is `std::shared_ptr` , while on older version is boost. I think you should use `pcl::make_shared`  to work fine in both cases.
```suggestion
    pc = pcl::make_shared<pcl::PointCloud<pcl::PointXYZRGB>>();    
```

Reply:
Thanks, that fixed the macos ci https://github.com/dic-iit/bipedal-locomotion-framework/pull/165/commits/bdfb6bb835f7048d8c76ae152b82362ff5a6f7e9 and https://github.com/dic-iit/bipedal-locomotion-framework/pull/165/commits/c404b4169b603093a6fcfeb6e23927004b518b16.